File: Common/AppStatusLib/src/main/java/com/yc/appstatuslib/utils/ThreadManagerUtils.java
package com.yc.appstatuslib.utils;



import java.lang.Thread.State;
import java.util.ArrayList;
import java.util.List;
import java.util.Set;

public final class ThreadManagerUtils {

    private static ThreadManagerUtils INSTANCE;

    public static ThreadManagerUtils getInstance() {
        if (INSTANCE == null) {
            synchronized (ThreadManagerUtils.class) {
                if (INSTANCE == null) {
                    INSTANCE = new ThreadManagerUtils();
                }
            }
        }
        return INSTANCE;
    }


    private ThreadManagerUtils() {

    }

    public int getThreadCount() {
        return Thread.getAllStackTraces().keySet().size();
    }

    public List<Thread> getRunningThread() {
        Set<Thread> threadSet = Thread.getAllStackTraces().keySet();
        Thread[] threads = (Thread[])threadSet.toArray(new Thread[threadSet.size()]);
        List<Thread> runnableThread = new ArrayList();
        Thread[] var4 = threads;
        int var5 = threads.length;
        for(int var6 = 0; var6 < var5; ++var6) {
            Thread thread = var4[var6];
            if (thread.getState() == State.RUNNABLE) {
                runnableThread.add(thread);
            }
        }
        return runnableThread;
    }

    public List<Thread> getBlockThread() {
        Set<Thread> threadSet = Thread.getAllStackTraces().keySet();
        Thread[] threads = (Thread[])threadSet.toArray(new Thread[threadSet.size()]);
        List<Thread> runnableThread = new ArrayList();
        Thread[] var4 = threads;
        int var5 = threads.length;
        for(int var6 = 0; var6 < var5; ++var6) {
            Thread thread = var4[var6];
            if (thread.getState() == State.BLOCKED) {
                runnableThread.add(thread);
            }
        }
        return runnableThread;
    }

    public List<Thread> getTimeWaitingThread() {
        Set<Thread> threadSet = Thread.getAllStackTraces().keySet();
        Thread[] threads = (Thread[])threadSet.toArray(new Thread[threadSet.size()]);
        List<Thread> runnableThread = new ArrayList();
        Thread[] var4 = threads;
        int var5 = threads.length;
        for(int var6 = 0; var6 < var5; ++var6) {
            Thread thread = var4[var6];
            if (thread.getState() == State.TIMED_WAITING) {
                runnableThread.add(thread);
            }
        }
        return runnableThread;
    }

    public List<Thread> getWaitingThread() {
        Set<Thread> threadSet = Thread.getAllStackTraces().keySet();
        Thread[] threads = (Thread[])threadSet.toArray(new Thread[threadSet.size()]);
        List<Thread> runnableThread = new ArrayList();
        Thread[] var4 = threads;
        int var5 = threads.length;
        for(int var6 = 0; var6 < var5; ++var6) {
            Thread thread = var4[var6];
            if (thread.getState() == State.WAITING) {
                runnableThread.add(thread);
            }
        }
        return runnableThread;
    }
}

File: CommonLib/AppStatusLib/src/main/java/com/yc/appstatuslib/utils/ThreadManagerUtils.kt
package com.yc.appstatuslib.utils

import java.util.*

/**
 * <pre>
 * @author yangchong
 * email  : yangchong211@163.com
 * time   : 2017/5/18
 * desc   : 线程工具类
 * revise :
</pre> *
 */
class ThreadManagerUtils private constructor() {

    val threadCount: Int
        get() = Thread.getAllStackTraces().keys.size

    val runningThread: List<Thread>
        get() {
            val threadSet: Set<Thread> = Thread.getAllStackTraces().keys
            val threads = threadSet.toTypedArray()
            val runnableThread: MutableList<Thread> = ArrayList()
            val var5 = threads.size
            for (i in 0 until var5) {
                val thread = threads[i]
                if (thread.state == Thread.State.RUNNABLE) {
                    runnableThread.add(thread)
                }
            }
            return runnableThread
        }

    val blockThread: List<Thread>
        get() {
            val threadSet: Set<Thread> = Thread.getAllStackTraces().keys
            val threads = threadSet.toTypedArray()
            val runnableThread: MutableList<Thread> = ArrayList()
            val length = threads.size
            for (i in 0 until length) {
                val thread = threads[i]
                if (thread.state == Thread.State.BLOCKED) {
                    runnableThread.add(thread)
                }
            }
            return runnableThread
        }

    val timeWaitingThread: List<Thread>
        get() {
            val threadSet: Set<Thread> = Thread.getAllStackTraces().keys
            val threads = threadSet.toTypedArray()
            val runnableThread: MutableList<Thread> = ArrayList()
            for (i in threads.indices) {
                val thread = threads[i]
                if (thread.state == Thread.State.TIMED_WAITING) {
                    runnableThread.add(thread)
                }
            }
            return runnableThread
        }

    val waitingThread: List<Thread>
        get() {
            val threadSet: Set<Thread> = Thread.getAllStackTraces().keys
            val threads = threadSet.toTypedArray()
            val runnableThread: MutableList<Thread> = ArrayList()
            for (i in threads.indices) {
                val thread = threads[i]
                if (thread.state == Thread.State.WAITING) {
                    runnableThread.add(thread)
                }
            }
            return runnableThread
        }

    companion object {
        private var INSTANCE: ThreadManagerUtils? = null
        @JvmStatic
        val instance: ThreadManagerUtils?
            get() {
                if (INSTANCE == null) {
                    synchronized(ThreadManagerUtils::class.java) {
                        if (INSTANCE == null) {
                            INSTANCE = ThreadManagerUtils()
                        }
                    }
                }
                return INSTANCE
            }
    }
}